wdl-lint/src/rules/comment_whitespace.rs

```diff
@@ -0,0 +1,182 @@
+        .with_highlight(span)
+        .with_fix("this comment must be preceded with two spaces")
+}
+
+/// Creates a diagnostic when the comment token is not followed by a single
+/// space.
+fn following_whitespace(span: Span) -> Diagnostic {
+    Diagnostic::note("comment delimiter should be followed by a single space")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix("follow this comment delimiter with a single space")
+}
+
+/// Creates a diagnostic when non-inline comment has insufficient indentation.
+fn insufficient_indentation(span: Span, expected: usize, actual: usize) -> Diagnostic {
+    Diagnostic::note("comment not sufficiently indented")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix(format!(
+            "this comment has {actual} levels of indentation. It should have {expected} levels of \
+             indentation."
+        ))
+}
+
+/// Creates a diagnostic when non-inline comment has excess indentation.
+fn excess_indentation(span: Span, expected: usize, actual: usize) -> Diagnostic {
+    Diagnostic::note("comment has too much indentation")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix(format!(
+            "this comment has {actual} levels of indentation. It should have {expected} levels of \
+             indentation."
+        ))
+}
+
+/// Detects improperly spaced comments.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct CommentWhitespaceRule;
+
+impl Rule for CommentWhitespaceRule {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Ensures that WDL comments have the proper spacing."
+    }
+
+    fn explanation(&self) -> &'static str {
+        "Comments on the same line as code should have 2 spaces before the # and one space before \
+         the comment text. Comments on their own line should match the indentation level around \
+         them and have one space between the # and the comment text. Keep in mind that even \
+         comments must be kept below the 90 character width limit."
+    }
+
+    fn tags(&self) -> TagSet {
+        TagSet::new(&[Tag::Spacing])
+    }
+}
+
+impl Visitor for CommentWhitespaceRule {
+    type State = Diagnostics;
+
+    fn document(
+        &mut self,
+        _: &mut Self::State,
+        reason: VisitReason,
+        _: &Document,
+        _: SupportedVersion,
+    ) {
+        if reason == VisitReason::Exit {
+            return;
+        }
+
+        // Reset the visitor upon document entry
+        *self = Default::default();
+    }
+
+    fn comment(&mut self, state: &mut Self::State, comment: &Comment) {
+        let re = Regex::new(r"^#+@?").unwrap();
+        if is_inline_comment(comment) {
+            // check preceding whitespace for two spaces
+            if let Some(prior) = comment.syntax().prev_sibling_or_token() {
+                if prior.kind() == SyntaxKind::Whitespace && prior.to_string() != "  " {
```

Comment:
I think this will miss the case when an inline comment isn't preceded by whitespace. e.g.
```
meta {# this is a problematic yet valid comment
}
```

Reply:
Good catch. That should be.

```suggestion
                if prior.kind() != SyntaxKind::Whitespace || prior.to_string() != "  " {
```